### PR TITLE
#77 Upgrade nuget Newtonsoft.Json to latest version 12.0.3

### PR DESCRIPTION
**Pull request by mmilitaru**
#284 Upgrade nuget AvalonEdit to latest version 6.0.1

### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [ ] I have provided test coverage for my change (where applicable)

### Description
#77 Upgrading all projects using NewtonSoft.Json to latest version 12.0.3
#284 Upgrading all projects using AvalonEdit to latest version 6.0.1

### CODE REVIEW

File: CDP4Addin/CDP4Addin.csproj [hunk 1]
```diff
@@ -118,8 +118,7 @@
       <EmbedInteropTypes>False</EmbedInteropTypes>
     </Reference>
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>False</Private>
```
Comment (lxatrhea) [inline CDP4Addin/CDP4Addin.csproj]: Should stay private.

File: BasicRdl/BasicRdl.csproj [hunk 2]
```diff
@@ -150,8 +150,7 @@
       <Private>False</Private>
     </Reference>
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>False</Private>
```
Comment (lxatrhea) [inline BasicRdl/BasicRdl.csproj]: Example of private = False indication that should imho stay that way.

**Comment by samatrhea** [comment]
I propose we add the newtonsoft json package to the setreferencecopylocal.ps1 script so that it is never copied and only occurs once in the solution, in the root folder of the app

File: CDP4BuiltInRules/CDP4BuiltInRules.csproj [hunk 3]
```diff
@@ -168,8 +168,7 @@
       <Private>False</Private>
     </Reference>
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
-      <Private>False</Private>
```
Comment (alexatrhea) [inline CDP4BuiltInRules/CDP4BuiltInRules.csproj]: false missed

**Comment by mmilitaru** [comment]
#77 The script has been added and ran to make NewtonSoft.Json as private in all projects, except IME at once.

**Pull request merged**